[guides/images] Streamline (old) image integration content (#3707)
withastro/docs: src/content/docs/en/guides/images.mdx

Astro provides several ways for you to use images on your site, whether they are stored locally inside your project, linked to remotely, or stored in a CMS or CDN!

:::note[`astro:assets` (Experimental - coming in v3.0)]

> The experimental `astro:assets` module will be built in to `astro@3.0`.

Reviewer: Experimental assets is technically already "built-in", it's just under a flag. Maybe "enabled by default" or something similar would be better?

> The experimental `astro:assets` module will be built in to `astro@3.0`.

Author: ```suggestion
The experimental `astro:assets` module will be enabled by default in `astro@3.0`.
```

## Astro's Image Integration

:::note[to be deprecated in v3.0]

> This integration will no longer be actively supported in Astro v3.0. 

Reviewer: Maybe `@astrojs/image` should be included in this paragraph so people know what integration we're talking about?
